extensions/sql-database-projects/src/controllers/projectController.ts

```diff
@@ -114,28 +117,82 @@ export class ProjectsController {
 	}
 
 	public closeProject(treeNode: BaseProjectTreeItem) {
-		const project = this.getProjectContextFromTreeNode(treeNode);
+		const project = ProjectsController.getProjectFromContext(treeNode);
 		this.projects = this.projects.filter((e) => { return e !== project; });
 		this.refreshProjectsTree();
 	}
 
-	public async buildProject(treeNode: BaseProjectTreeItem): Promise<void> {
+	/**
+	 * Builds a project, producing a dacpac
+	 * @param treeNode a treeItem in a project's hierarchy, to be used to obtain a Project
+	 * @returns path of the built dacpac
+	 */
+	public async buildProject(treeNode: BaseProjectTreeItem): Promise<string>;
+	/**
+	 * Builds a project, producing a dacpac
+	 * @param project Project to be built
+	 * @returns path of the built dacpac
+	 */
+	public async buildProject(project: Project): Promise<string>;
+	public async buildProject(context: Project | BaseProjectTreeItem): Promise<string | undefined> {
```

Comment:
are there three of these intentionally?

Reply:
Yup, it lets me give better docstrings for the individual cases.  I don't love it, but it's how method overloads in typescript work.